app/src/main/java/org/projectbuendia/client/ui/SnackBar.java

```diff
@@ -0,0 +1,287 @@
+            Map<String, ?>>> childData, int childLayout, String[] childFrom, int[] childTo) {
+            super(context, groupData, groupLayout, groupFrom, groupTo, childData, childLayout,
+                childFrom, childTo);
+            this.snackBar = snackBar;
+        }
+
+        @Override
+        public View getChildView(int groupPosition, int childPosition, boolean isLastChild,
+                                 View convertView, ViewGroup parent) {
+            View newView = super.getChildView(groupPosition, childPosition, isLastChild,
+                convertView, parent);
+            addActonHandler(newView);
+            addDismissHandler(newView);
+            return newView;
+        }
+
+        @Override
+        public View getGroupView(int groupPosition, boolean isExpanded, View convertView,
+                                ViewGroup parent) {
+            View newView = super.getGroupView(groupPosition, isExpanded, convertView, parent);
+            addActonHandler(newView);
+            addDismissHandler(newView);
+            return newView;
+        }
+
+        private void addActonHandler(View newView) {
+            TextView idView = (TextView) newView.findViewById(R.id.snackbar_id);
+            String id = idView.getText().toString();
+            TextView priorityView = (TextView) newView.findViewById(R.id.snackbar_priority);
+            String priority = priorityView.getText().toString();
+            Message m = mMessagesList.get(new MessageKey(Integer.parseInt(id), Integer.parseInt(priority)));
+            TextView actionButton = (TextView) newView.findViewById(R.id.snackbar_action);
+            String action = actionButton.getText().toString();
+            if ((actionButton != null) && (!action.isEmpty())) {
+                if((m != null) && (m.actionHandler != null)){
+                    actionButton.setOnClickListener(m.actionHandler);
+                }
+            }
+        }
+
+        private void addDismissHandler(View newView) {
+            TextView idView = (TextView) newView.findViewById(R.id.snackbar_id);
+            final String id = idView.getText().toString();
+            TextView priorityView = (TextView) newView.findViewById(R.id.snackbar_priority);
+            final String priority = priorityView.getText().toString();
+            ImageView dismissButton = (ImageView) newView.findViewById(R.id.snackbar_dismiss);
+            dismissButton.setOnClickListener(new View.OnClickListener() {
+                @Override public void onClick(View v) {
+                    snackBar.dismiss(
+                        new MessageKey(Integer.parseInt(id), Integer.parseInt(priority))
+                    );
+                }
+            });
+        }
+    }
+
+    private class MessageKey implements Comparable<MessageKey> {
+
+        protected int id;
+        protected int priority;
+
+        public MessageKey(int id, int priority){
```

Comment:
Is there a reason why `priority` is part of the MessageKey? I think it would be more efficient if a MessageKey was just a unique integer ID. For what it's worth, Android uses unique integer IDs for notifications.